lib/rust/parser/jni/src/lib.rs

```diff
@@ -165,34 +146,11 @@ pub extern "system" fn Java_org_enso_syntax2_Parser_getMetadata(
     }
 }
 
-/// Allocate a new parser state object. The returned value should be passed to `freeState` when no
-/// longer needed.
-#[allow(unsafe_code, clippy::box_default)]
-#[no_mangle]
-pub extern "system" fn Java_org_enso_syntax2_Parser_allocState(
-    _env: JNIEnv,
-    _class: JClass,
-) -> u64 {
-    Box::into_raw(Box::new(State::default())) as _
-}
-
-/// Free the resources owned by the state object.
-///
-/// # Safety
-///
-/// The input MUST have been returned by `allocState`, and MUST NOT have previously been passed to
-/// `freeState`.
+/// Free the thread's parsing resources.
 #[allow(unsafe_code)]
 #[no_mangle]
-pub extern "system" fn Java_org_enso_syntax2_Parser_freeState(
-    _env: JNIEnv,
-    _class: JClass,
-    state: u64,
-) {
-    if state != 0 {
-        let state = unsafe { Box::from_raw(state as usize as *mut State) };
-        drop(state);
-    }
+pub extern "system" fn Java_org_enso_syntax2_Parser_freeBuffers(_env: JNIEnv, _class: JClass) {
```

Comment:
Is it still necessary to `freeBuffers` from Java usage? I don't see it happening in all the places that do some parsing...

Reply:
I think the Rust library should offer a way to free the resources, but as currently used there's not much reason to do so (at most one state is maintained per thread, and they can simply be freed when the process exits).